private/nonlconAnalytic.m

```diff
@@ -1,19 +1,22 @@
-function [c,ceq,gradc,gradceq] = nonlconAnalytic(x,tolIsotropy,gMax,integralConstraint,targetTensor,tolMaxwell,signs, useMaxNorm)
+function [c,ceq,gradc,gradceq] = nonlconAnalytic(x,tolIsotropy,gMax,integralConstraint,targetTensor,tolMaxwell,signs, useMaxNorm, motionCompensation, dt)
 %NONLCONANALYTIC evaluates the nonlinear constraints and their gradients.
 
 q = x(1:end-1);
 s = x(end);
 N = length(q)/3;
 Q = reshape(q,[N 3]);
 
+% Remove action on axes that are turned off
+Q = Q .* ((diag(targetTensor))>0)';
+
```

Comment:
This feels ad hoc. Is it necessary?

Reply:
Sorry for the ad-hocness, but from some basic tests I feel confident that it improves performance in rank(B) < 3. If you like, remove it from this update and let me do some more tests before inclusion.

Reply:
Will remove for the moment. Make feature request on improved handling of disables axes.